pkg/shared/client.go

```diff
@@ -65,3 +72,20 @@ func getConfigFromServiceAccount() *rest.Config {
 	}
 	return config
 }
+
+// eventRecorder returns an EventRecorder type that can be
+// used to post Events to different object's lifecycles.
+func getEventRecorder() record.EventRecorder {
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(
+		&typedcorev1.EventSinkImpl{
+			Interface: Client.Clientset.CoreV1().Events(""),
+		},
+	)
+	recorder := eventBroadcaster.NewRecorder(
+		scheme.Scheme,
+		v1.EventSource{Component: "kubedirector"},
```

Comment:
is there a constant definition we can  use here instead of the "kubedirector" string literal?

Reply:
we have an environment variable called OPERATOR_NAME. I can use that..